src/coffee/directives/api/models/parent/base-polys-parent-model.coffee

```diff
@@ -20,33 +20,33 @@ angular.module('uiGmapgoogle-maps.directives.api.models.parent')
           @firstTime = true
           @$log.info @
 
-          #@watchOurScope(scope)
+          # @watchOurScope(scope)
           @createChildScopes()
 
         #watch this scope(Parent to all Models), these updates reflect expression / Key changes
         #thus they need to be pushed to all the children models so that they are bound to the correct objects / keys
-  #      watch: (scope, name, nameKey) =>
-  #        scope.$watch name, (newValue, oldValue) =>
-  #          if (newValue != oldValue)
-  #            maybeCanceled =  null
-  #            @[nameKey] = if _.isFunction newValue then newValue() else newValue
-  #
-  #            _async.promiseLock @, uiGmapPromise.promiseTypes.update, "watch #{name} #{nameKey}"
-  #            , ((canceledMsg) -> maybeCanceled = canceledMsg)
-  #            , =>
-  #              _async.each @plurals.values(), (model) =>
-  #                model.scope[name] = if @[nameKey] == 'self' then model else model[@[nameKey]]
-  #                maybeCanceled
-  #              , _async.chunkSizeFrom scope.chunk
-
+        # watch: (scope, name, nameKey) =>
+        #   scope.$watch name, (newValue, oldValue) =>
+        #     if (newValue != oldValue)
+        #       maybeCanceled =  null
+        #       @[nameKey] = if _.isFunction newValue then newValue() else newValue
+        #
+        #       _async.promiseLock @, uiGmapPromise.promiseTypes.update, "watch #{name} #{nameKey}"
+        #       , ((canceledMsg) -> maybeCanceled = canceledMsg)
+        #       , =>
+        #        _async.each @plurals.values(), (model) =>
+        #          model.scope[name] = if @[nameKey] == 'self' then model else model[@[nameKey]]
+        #          maybeCanceled
+        #        , _async.chunkSizeFrom scope.chunk
 
         watchModels: (scope) =>
-          scope.$watchCollection 'models', (newValue, oldValue) =>
+          scope.$watch 'models', (newValue, oldValue) =>
```

Comment:
Probably, but watchCollection was not firing since the childScope is essentially detached from the model the only way we can get the update is via models. So if we want simple changes like visible and editable they get dropped. But if the user knows to replace the model object instance then watchCollection should catch it. 

I guess we could set it up to watchCollection or watch based on some setting. I didn't really care to go that deep; and instead just fix the issue.